pkg/apiserver/vshn/k8up/k8up.go

```diff
@@ -39,11 +39,15 @@ type ConcreteSnapshotHandler struct {
 }
 
 // Get returns the snapshot with the given ID.
-func (c *ConcreteSnapshotHandler) Get(ctx context.Context, id, instanceNamespace string) (*k8upv1.Snapshot, error) {
+func (c *ConcreteSnapshotHandler) Get(ctx context.Context, id, instanceNamespace string, scClient client.Client) (*k8upv1.Snapshot, error) {
 
 	snapshot := &k8upv1.Snapshot{}
-
-	err := c.client.Get(ctx, client.ObjectKey{Namespace: instanceNamespace, Name: id}, snapshot)
+	var err error
+	if scClient != nil {
```

Comment:
at this point in code is it not yet clear if you are in converged or non-converged mode?

Reply:
The APIServer itself doesn't know and doesn't care if it's in converged or non-converged mode